src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,40 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private final DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder result = new StringBuilder("Report for period ");
+        result.append(getFormattedData(dateFrom, "-", dateTo));
+        for (String name : names) {
+            int sumSalary = 0;
+            for (String value : data) {
+                String[] dataSplitted = value.split(" ");
+                if (isWithinPeriod(dataSplitted[0], dateFrom, dateTo)
+                        && name.equals(dataSplitted[1])) {
+                    sumSalary += Integer.parseInt(dataSplitted[2])
+                            * Integer.parseInt(dataSplitted[3]);
+                }
+            }
+            result.append(System.lineSeparator())
+                    .append(getFormattedData(name, "-", Integer.toString(sumSalary)));
+        }
+        return result.toString();
+    }
+
+    private String getFormattedData(String...data) {
```

Comment:
The method 'getFormattedData' is unnecessary and can be removed. You can directly use 'String.join' wherever you need to concatenate strings with spaces.